### ISSUE TITLE
[Request] Add option to disable viewshake / camera shake

### ISSUE THREAD

**Issue by dizzy-egg** [OldUnreal/UnrealTournamentPatches#280]
UT uses a viewshake method to shake the player's camera to simulate things like earthquakes or very high damage from high falls.

Unfortunately this (like the related screenflashes effect) can be a trigger for people with epilepsy or vestibular issues like vertigo.

Adding an option to disable viewshake would be extremely helpful.

**Comment by BerserkerBG** [comment]
I think this is a great idea, however, I do think we should also include an option to disable the fire screen shake from weapon fire as well.

An idea would be to add 2 additional options under the Game section in Preferences:

![viewshake](https://github.com/user-attachments/assets/ad461ff4-0746-4421-a277-255a8a25c00b)

Weapon fire check will deal with the viewshake when firing your weapons. Damaged shake will deal with viewshake when taking damage like bullets or landing from high falls (should also deal with earthquakes as well).

**Comment by juliomonk** [comment]
I agree, there should be an option to change the screen shake and also screen flash as an accesibility option. Right now you can only turn the screen flash off in the "UnrealTournament.ini" system file. It's still a cool effect though, perhaps can be modified to have half the intensity/opacity?

**Comment by NahuelGeno** [comment]
I think it's a fantastic idea, I'm looking forward to that option.

**Comment by sn260591** [comment]
Alternatively, instead of a checkbox, there can be a numeric value that limits the maximum camera shake (or disabling it completely when set to 0). Some game modes and mutators increase the player's maximum health to several thousand, and from huge damage (over 1000) the camera begins to spin wildly.
